Make test result doubles accept event log

jml opened the pull request:
I was looking around at the test result code in preparation for using it with flocker. While I was there, I noticed a heap of lint in flake8, so I've fixed it.

There's a small semantic change also. Add an `event_log` parameter to all of the test result doubles, and use that if it's provided. This spares code that uses them from having to know about `_log`.

<!-- Reviewable:start -->
[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/testing-cabal/testtools/157)
<!-- Reviewable:end -->

thomir commented:
Reviewed 3 of 3 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

<sup>**[testtools/testresult/real.py, line 387 \[r1\]](https://reviewable.io:443/reviews/testing-cabal/testtools/157#-K2999LzYR-Fpu_0aLm-)** ([raw file](https://github.com/testing-cabal/testtools/blob/bb220ee20cae3f285d8a2f600aabeb89a3a4349a/testtools/testresult/real.py#L387)):</sup>
It's not clear to me what 'xs' stands for? Perhaps consider changing this name to be more descriptive?
<!-- Reviewable comment -K2999M-tWf7uxsQJ5fI:-818872525 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/testing-cabal/testtools/157)
<!-- Sent from Reviewable.io -->

jml commented:
Review status: 2 of 3 files reviewed at latest revision, 1 unresolved discussion.

---

<sup>**[testtools/testresult/real.py, line 387 \[r1\]](https://reviewable.io:443/reviews/testing-cabal/testtools/157#-K2999LzYR-Fpu_0aLm-)** ([raw file](https://github.com/testing-cabal/testtools/blob/bb220ee20cae3f285d8a2f600aabeb89a3a4349a/testtools/testresult/real.py#L387)):</sup>
Changed.

FWIW, `xs` is a Haskell idiom for a list of arbitrary things.
<!-- Reviewable comment -K2BtAkd5D6TcHDoxEtw:1820367474 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/testing-cabal/testtools/157)
<!-- Sent from Reviewable.io -->

thomir commented:
LGTM - ship it ;)

rbtcollins commented:
Review status: 2 of 3 files reviewed at latest revision, 8 unresolved discussions.

---

<sup>**[testtools/testresult/real.py, line 358 \[r2\]](https://reviewable.io:443/reviews/testing-cabal/testtools/157#-K2Dghs386g6B6r_Pv5K)** ([raw file](https://github.com/testing-cabal/testtools/blob/a854c02aa7564d9ff9ce5ea1d795d6f61072b13e/testtools/testresult/real.py#L358)):</sup>
I'm confused by these reflowings - they're not harmful, but why are you doing them?
<!-- Reviewable comment -K2Dghs4M1UVHjW5cbtI:836332888 -->

---

<sup>**[testtools/testresult/real.py, line 387 \[r2\]](https://reviewable.io:443/reviews/testing-cabal/testtools/157#-K2De-zmwupKHz2ScroD)** ([raw file](https://github.com/testing-cabal/testtools/blob/7c7f747f1092c8845a96041fff85d685703339e0/testtools/testresult/real.py#L387)):</sup>
This should be _strict_map, since you're not intending it to be in the public API AFAICT.

And please don't call it SOMETHING_map if it doesn't abide by maps' signature (which it doesn't).
<!-- Reviewable comment -K2De-zmwupKHz2ScroE:-1793596446 -->

---

<sup>**[testtools/testresult/real.py, line 389 \[r2\]](https://reviewable.io:443/reviews/testing-cabal/testtools/157#-K2Dh5tgFNy1wXACfw4E)** ([raw file](https://github.com/testing-cabal/testtools/blob/a854c02aa7564d9ff9ce5ea1d795d6f61072b13e/testtools/testresult/real.py#L389)):</sup>
renaming this is possibly going to break people - and its not clearly marked as private, so I'd like to keep an alias to it around.

The change from *args + **kwargs to function, items is making your new thing incompatible with actual map (which can take many sequences), so the name change does needs to happen if you're making that change, so please keep an unaltered version around - perhaps with a deprecation warning or something.
<!-- Reviewable comment -K2Dh5tgFNy1wXACfw4F:-1551654209 -->

---

<sup>**[testtools/testresult/real.py, line 431 \[r2\]](https://reviewable.io:443/reviews/testing-cabal/testtools/157#-K2DhmsctrbOJnRGYzig)** ([raw file](https://github.com/testing-cabal/testtools/blob/7c7f747f1092c8845a96041fff85d685703339e0/testtools/testresult/real.py#L431)):</sup>
I don't super object, but I also don't understand these reformattings. If the function was nested I would object because it becomes very cramped on the right hand side. I'm guessing you ran a naive pep-8 thingy over the code?
<!-- Reviewable comment -K2DhmsctrbOJnRGYzih:796244191 -->

---

<sup>**[testtools/testresult/real.py, line 702 \[r2\]](https://reviewable.io:443/reviews/testing-cabal/testtools/157#-K2Di6plxVztUiLt-nPO)** ([raw file](https://github.com/testing-cabal/testtools/blob/7c7f747f1092c8845a96041fff85d685703339e0/testtools/testresult/real.py#L702)):</sup>
This is an example of where I'd start to object: this openning bracket aligned style quickly becomes unmanagable. Hanging indents don't, and are PEP8 (if that is whats driving this...)
<!-- Reviewable comment -K2Di6plxVztUiLt-nPP:242970980 -->

---

<sup>**[testtools/tests/test_testresult.py, line 227 \[r2\]](https://reviewable.io:443/reviews/testing-cabal/testtools/157#-K2Dj-8JrfcVMLumUl_p)** ([raw file](https://github.com/testing-cabal/testtools/blob/7c7f747f1092c8845a96041fff85d685703339e0/testtools/tests/test_testresult.py#L227)):</sup>
I find the application of top level VWS rules around nested classes impedes readability - it conflicts with the VWS rules for inside functions which advise sparing use. I'd rather we turned the rule off entirely for that rather than blindly follow it.
<!-- Reviewable comment -K2Dj-8JrfcVMLumUl_q:-1376104182 -->

---

<sup>**[testtools/tests/test_testresult.py, line 533 \[r2\]](https://reviewable.io:443/reviews/testing-cabal/testtools/157#-K2DjG5AvWtu9ZAHJ7xV)** ([raw file](https://github.com/testing-cabal/testtools/blob/7c7f747f1092c8845a96041fff85d685703339e0/testtools/tests/test_testresult.py#L533)):</sup>
Similarly here, the massive use of height impedes comprehension.
<!-- Reviewable comment -K2DjG5AvWtu9ZAHJ7xW:503243862 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/testing-cabal/testtools/157)
<!-- Sent from Reviewable.io -->

rbtcollins commented:
Ah, I just noticed - flake8. So my experience with flake8 is that you have to turn a bunch of its checks off to have clean code. Otherwise you get *consistently bad code. pep8 was *never* intended for use outside the standard library, and while I do subscribe to the gofmt philosophy, flake8 ain't that: because a) it doesn't pretty print for you, so when the rules are in the way you can't issue new rules and have it Just Work, and b) the rules haven't been hammered out in the expectation that everyone will blindly apply them.

jml commented:
Hi Rob,

I know that PEP 8 was not written with the intent of being used outside the standard library. I've read the darn thing several times over.

Nevertheless, 
* it _is_ used very widely outside the stdlib
* it's the documented coding standard for testtools

My experience with flake8 is almost directly contrary to yours.

jml
<!-- Reviewable comment -K2DmL5C02IAdShvcMwh:1130071237 -->

---

Review status: 1 of 3 files reviewed at latest revision, 8 unresolved discussions.

---

<sup>**[testtools/testresult/real.py, line 358 \[r2\]](https://reviewable.io:443/reviews/testing-cabal/testtools/157#-K2Dghs386g6B6r_Pv5K)** ([raw file](https://github.com/testing-cabal/testtools/blob/a854c02aa7564d9ff9ce5ea1d795d6f61072b13e/testtools/testresult/real.py#L358)):</sup>
Standard column limits. See summary comment.
<!-- Reviewable comment -K2DndsVZi5CwA_rFKxZ:-1430067636 -->

---

<sup>**[testtools/testresult/real.py, line 387 \[r2\]](https://reviewable.io:443/reviews/testing-cabal/testtools/157#-K2De-zmwupKHz2ScroD)** ([raw file](https://github.com/testing-cabal/testtools/blob/7c7f747f1092c8845a96041fff85d685703339e0/testtools/testresult/real.py#L387)):</sup>
Agree re `_` prefix. Have changed to `_strict_map` and changed `items` to `*sequences`, which matches the `map` signature.
<!-- Reviewable comment -K2DniRU7kpWyjBmnIG_:1631568186 -->

---

<sup>**[testtools/testresult/real.py, line 389 \[r2\]](https://reviewable.io:443/reviews/testing-cabal/testtools/157#-K2Dh5tgFNy1wXACfw4E)** ([raw file](https://github.com/testing-cabal/testtools/blob/a854c02aa7564d9ff9ce5ea1d795d6f61072b13e/testtools/testresult/real.py#L389)):</sup>
Added `domap` alias with deprecation warning, since it's an inappropriate function to be providing publicly from this module, and arguably out-of-scope for our public API altogether. Maybe we can add something to extras if we care enough?
<!-- Reviewable comment -K2DnMnPSmQ96G1P7tuf:-329454764 -->

---

<sup>**[testtools/testresult/real.py, line 431 \[r2\]](https://reviewable.io:443/reviews/testing-cabal/testtools/157#-K2DhmsctrbOJnRGYzig)** ([raw file](https://github.com/testing-cabal/testtools/blob/7c7f747f1092c8845a96041fff85d685703339e0/testtools/testresult/real.py#L431)):</sup>
Yes, flake8.
<!-- Reviewable comment -K2Rwia9D9doQgxmaWCr:-645183659 -->

---

<sup>**[testtools/testresult/real.py, line 702 \[r2\]](https://reviewable.io:443/reviews/testing-cabal/testtools/157#-K2Di6plxVztUiLt-nPO)** ([raw file](https://github.com/testing-cabal/testtools/blob/7c7f747f1092c8845a96041fff85d685703339e0/testtools/testresult/real.py#L702)):</sup>
Yeah, I don't like this version either. Changed to putting the arguments as hanging indent with first arg on a newline, which is PEP 8 compliant and addresses that issue.
<!-- Reviewable comment -K2DocNPxREMv90xp5p4:411209310 -->

---

<sup>**[testtools/tests/test_testresult.py, line 227 \[r2\]](https://reviewable.io:443/reviews/testing-cabal/testtools/157#-K2Dj-8JrfcVMLumUl_p)** ([raw file](https://github.com/testing-cabal/testtools/blob/7c7f747f1092c8845a96041fff85d685703339e0/testtools/tests/test_testresult.py#L227)):</sup>
I genuinely find the version with vertical whitespace easier to read—I'm not blindly following anything. However, I think the readability gain is marginal, so I've reverted.
<!-- Reviewable comment -K2DoHuFwCbRi_07OtPc:-1767119768 -->

---

<sup>**[testtools/tests/test_testresult.py, line 533 \[r2\]](https://reviewable.io:443/reviews/testing-cabal/testtools/157#-K2DjG5AvWtu9ZAHJ7xV)** ([raw file](https://github.com/testing-cabal/testtools/blob/7c7f747f1092c8845a96041fff85d685703339e0/testtools/tests/test_testresult.py#L533)):</sup>
Ah really? It's the opposite for me. Also makes it less likely to miss something, esp. if they're alpha sorted. 
<!-- Reviewable comment -K2Dmey3GOEbx4MNIlHE:-1133675099 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/testing-cabal/testtools/157)
<!-- Sent from Reviewable.io -->
